src/unicode.cpp

```diff
@@ -101,25 +101,28 @@ namespace ada::unicode {
    * @see https://github.com/nodejs/node/blob/main/src/node_url.cc#L226
    */
   std::string percent_encode(const std::string_view input, const uint8_t character_set[]) {
-    std::string result{};
+    auto pointer = std::find_if(input.begin(), input.end(), [character_set](const char c) {
+      return character_sets::bit_at(character_set, c);
+    });
+    // Optimization: Don't iterate if percent encode is not required
+    if (pointer == input.end()) { return std::string(input); }
```

Comment:
That's good. Fast in the common case, and probably just as fast as before in the pessimistic case.